src/ui/pages/AssetInfo/AssetHistory.tsx

```diff
@@ -0,0 +1,272 @@
+    transfer.quantity,
+    getDecimals({
+      asset: fungible,
+      chain: createChain(action.transaction.chain),
+    })
+  );
+
+  const normalizedAmountAction =
+    transfer.price === null || transfer.price === undefined
+      ? null
+      : normalizedQuantity.multipliedBy(transfer.price);
+
+  const actionTitle = `${actionType} at ${formattedPrice}`;
+  const actionDatetime = dateFormatter.format(new Date(action.datetime));
+  const actionBalance = `${isIncoming ? '+' : minus}${formatTokenValue(
+    normalizedQuantity,
+    asset.symbol,
+    { notation: normalizedQuantity.gte(100000) ? 'compact' : undefined }
+  )}`;
+  const actionValue = normalizedAmountAction
+    ? formatCurrencyValue(normalizedAmountAction, 'en', currency)
+    : null;
+
+  return (
+    <>
+      <KeyboardShortcut
+        combination="backspace"
+        onKeyDown={handleDialogDismiss}
+      />
+      <UnstyledButton onClick={handleDialogOpen} className={styles.historyItem}>
+        <div className={styles.historyItemBackdrop} />
+        <HStack
+          gap={12}
+          justifyContent="space-between"
+          style={{ position: 'relative', paddingBlock: 12 }}
+        >
+          <VStack gap={0} style={{ justifyItems: 'start' }}>
+            <UIText kind="body/regular">{actionTitle}</UIText>
+            <UIText kind="small/regular" color="var(--neutral-500)">
+              {actionDatetime}
+            </UIText>
+          </VStack>
+          <VStack gap={0} style={{ justifyItems: 'end' }}>
+            <UIText
+              kind="body/regular"
+              color={isIncoming ? 'var(--positive-500)' : 'currentColor'}
+            >
+              {actionBalance}
+            </UIText>
+            <UIText kind="small/regular" color="var(--neutral-500)">
+              {actionValue}
+            </UIText>
+          </VStack>
+        </HStack>
+      </UnstyledButton>
+      <CenteredDialog
+        ref={dialogRef}
+        containerStyle={{ backgroundColor: 'var(--neutral-100)' }}
+        renderWhenOpen={() => (
+          <>
+            <Button
+              kind="ghost"
+              value="cancel"
+              size={36}
+              style={{
+                width: 36,
+                padding: 8,
+                position: 'absolute',
+                top: 16,
+                left: 8,
+              }}
+              onClick={handleDialogDismiss}
+            >
+              <ArrowLeftIcon style={{ width: 20, height: 20 }} />
+            </Button>
+            <ActionDetailedView
+              action={action}
+              networks={networks}
+              address={address}
+            />
+          </>
+        )}
+      />
+    </>
+  );
+}
+
+export function AssetHistory({
+  assetId,
+  address,
+  assetFullInfo,
+}: {
+  assetId: string;
+  address: string;
+  assetFullInfo?: AssetFullInfo;
+}) {
+  const { networks } = useNetworks();
+  const { singleAddressNormalized, ready } = useAddressParams();
```

Comment:
I think `address` and `singleAddressNormalized` are being mixed up. It seems `address` from props should be enough